### DISCUSSION TITLE
REM Pressure Transducer PSI/volt value

### DISCUSSION THREAD

**Discussion by johnny2678** [tagyoureit/nodejs-poolController#509] [REM] [answered]
If I have[ this 80psi transducer](https://www.amazon.com/gp/product/B07KJHJNFQ/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&psc=1) hooked up to a 0-10v BAS input, what's the correct PSI/volt value?  

is the voltage denominator the input voltage or output voltage?

80/5 = 16? 
or
80/(4.5-0.5) = 20?

![image](https://user-images.githubusercontent.com/8219743/176028454-6266bdf4-ee86-4d20-a549-d0863a1ddb4b.png)

**Comment by johnny2678** [comment]
filter pressure dial is broken so not sure which reading is right.  16 obviously yields a lower PSI value.

**Reply by orrious** [reply]
Your input voltage should be 5v to that sensor.  That sensor outputs 4.5v @80psi.  So 80/4.5=17.7777psi per volt.  You should probably get a sensor with a lower range if you want more accuracy.

I have not seen my system very high, maybe 15psi or so.  What is the highest other folks have seen theirs?  Would a 30psi sensor suffice?

**Reply by johnny2678** [reply]
Nice. thx!

Just made the change and maxed out the pump.  It spiked to 28psi before settling on 26-27

![image](https://user-images.githubusercontent.com/8219743/176035665-b1fea4c4-2388-4914-a03b-f9182bb9d404.png)

30 wouldn't leave a lot of wiggle room in my system.  Maybe 50?

**Reply by tagyoureit** [reply]
From the Amazon webpage:
> 0.5 is 0psi 4.5 is the maximum (80psi) and is a linear curve .

You do need to take the starting PSI into consideration.  80/(4.5-.5) = 20 PSI per Volt.  

I've had [this 0-100psi](https://www.ebay.com/itm/261260635816?hash=item3cd458eea8:g:XVYAAOSwxBdZgiGA) sensor for years.  My PSI gets up to high 20's when the Jets are on full blast.

**Reply by johnny2678** [reply]
uh oh... so is it

80/(4.5 - 0.5) = 20

this is at odds with what @orrious just posted above...

80/4.5 = 17.777

hmmm

**Reply by tagyoureit** [reply]
The formula is `device.values.psi = parseFloat(Math.max(0, ((device.values.voltage - device.options.inducerOffset) * device.options.psiPerVolt))`.

So,
4.5v:  (4.5-.5)*20 = 80psi
0.5v: (.5-.5)*20 = 0psi
2.021v: (2.021-.5)*20 = 30.42v

If it was 17.777 psi/v then you would max out at (4.5-.5)*17.777=71.11psi.

**Reply by orrious** [reply]
I did not originally account for the 0.5v offset.  @tagyoureit is correct. 80/(4.5-.5) = 20 PSI per Volt.

**Comment by johnny2678** [comment, marked as answer]
from @tagyoureit 
> The formula is `device.values.psi = parseFloat(Math.max(0, ((device.values.voltage - device.options.inducerOffset) * device.options.psiPerVolt))`.
> 
> So, 4.5v: (4.5-.5)*20 = 80psi 0.5v: (.5-.5)*20 = 0psi 2.021v: (2.021-.5)*20 = 30.42v
> 
> If it was 17.777 psi/v then you would max out at (4.5-.5)*17.777=71.11psi.
